client/src/partials/accounts/accounts.js

```diff
@@ -2,192 +2,325 @@ angular.module('bhima.controllers')
-
-    var formatAccount = {
-      account_type_id: account.type.id,
-      account_number: account.number,
-      account_txt: account.title,
-      is_asset: account.is_asset,
-      is_ohada: account.is_ohada,
-      is_charge : account.is_charge,
-      is_used_budget: account.is_used_budget,
-      cc_id   : account.cc_id,
-      pc_id   : account.pc_id,
-      enterprise_id: appstate.get('enterprise').id,
-      parent: account.parent,
-      classe: account.number.substr(0,1),
-      reference_id : account.reference_id,
-      is_brut_link : account.is_brut_link
-    };
-
-    connect.post('account', [formatAccount])
-    .then(refreshAccountList)
-    .then(function () {
-      messenger.success($translate.instant('CONFIG_ACCOUNTING.SAVE_SUCCES'));
-      $scope.newAccount = {};
-      session.state = 'display';
-    });
+  function checkConditionAsset (){
+    if(!vm.newAccount.account_type_id) {return;}
+    var res = accountTypeService.getAccountType(vm.accountTypes, vm.newAccount.account_type_id);
+    return res.type == 'balance' && !vm.newAccount.parent;
   }
 
-  function updateState(newState) {
-    session.state = newState;
+  function checkConditionCharge (){
+    if(!vm.newAccount.account_type_id) {return;}
+    var res = accountTypeService.getAccountType(vm.accountTypes, vm.newAccount.account_type_id);
+    return res.type == 'income/expense' && !vm.newAccount.parent;
   }
 
-  function formatCenter (c) {
-    return '' + c.text;
+  function checkConditionProfitCenter (){
+    if(!vm.newAccount.account_type_id) {return;}
+    var res = accountTypeService.getAccountType(vm.accountTypes, vm.newAccount.account_type_id);
+    return res.type == 'income/expense' && vm.newAccount.is_charge == 0;
   }
 
-  $scope.discareCC = function () {
-    $scope.newAccount.cc_id = null;
-  };
+  function checkConditionCostCenter (){
+    if(!vm.newAccount.account_type_id) {return;}
+    var res = accountTypeService.getAccountType(vm.accountTypes, vm.newAccount.account_type_id);
+    return res.type == 'income/expense' && vm.newAccount.is_charge == 1;
+  }
 
-  $scope.discarePC = function () {
-    $scope.newAccount.pc_id = null;
-  };
+  function checkConditionType (){
+    return !vm.newAccount.parent;
+  }
 
-  $scope.getAccount = function (account) {
-    $scope.editAccount = null;
-    session.state = 'edit';
-    $scope.editAccount = account;
-  };
+  function format(a) {
+    return [a.account_number, a.account_txt].join(' :: ');
+  }
 
-  $scope.format = function format(account) {
-    return [account.account_number, account.account_txt].join(' :: ');
-  };
+  function formatReference (r) {
+    return r.ref + ' :: ' + r.text;
+  }
 
-  $scope.formatRef = function formatRef(reference) {
-    return [reference.ref, reference.text].join(' :: ');
-  };
+  function checkConditionPanelBilan() {
+    var res = null;
+    if(vm.newAccount.account_type_id) {res = accountTypeService.getAccountType(vm.accountTypes, vm.newAccount.account_type_id);}
```

Comment:
This is really long to put on a single line.  Could you format it better by putting it on multiple lines?